openshift/templates/postgres/deploy.yaml

```diff
@@ -0,0 +1,160 @@
+kind: Template
+apiVersion: template.openshift.io/v1
+metadata:
+  name: gdx-agreements-tracker-api-postgres-deployment
+  resourceVersion: "v1.0.0"
+  annotations:
+    openshift.io/display-name: Deployment GDX agreements tracker postgres database.
+    description: Deployment for GDX agreements tracker postgres database.
+    tags: pmo,api,gdx,agreements-tracker,postgres
+parameters:
+  - name: APP_NAME
+    displayName: "App Name"
+    description: "The name of the application (grouped)."
+    required: true
+    value: "gdx-agreements-tracker"
+  - name: ROLE_NAME
+    displayName: "Component Name"
+    description: "The name of the application role."
+    required: true
+    value: "api"
+  - name: PROJECT_NAMESPACE
+    displayName: "OpenShift Project Namespace"
+    description: "The namespace of the OpenShift project containing the application."
+    required: true
+    value: "acd38d"
+  - name: ENV_NAME
+    displayName: "Environment name"
+    description: "The name for this environment [dev, test, prod, tools]"
+    required: true
+    value: "dev"
+
+  - name: APP_PORT
+    displayName: "Application Port"
+    description: "The port used to communicate with the Pod"
+    value: "5432"
+  - name: REPLICAS
+    displayName: "Pod Replicas"
+    description: "Number of pod replicas to run"
+    value: "1"
+  - name: CPU_REQUEST
+    displayName: "CPU Request"
+    description: "Starting amount of CPU the container can use."
+    value: "100m"
+  - name: CPU_LIMIT
+    displayName: "CPU Limit"
+    description: "Maximum amount of CPU the container can use."
+    value: "1"
+  - name: MEMORY_REQUEST
+    displayName: "Memory Request"
+    description: "Starting amount of memory the container can use."
+    value: "100Mi"
+  - name: MEMORY_LIMIT
+    displayName: "Memory Limit"
+    description: "Maximum amount of memory the container can use."
+    value: "2Gi"
+
+objects:
+  - kind: Deployment
+    apiVersion: apps/v1
+    metadata:
```

Comment:
where is the annotation to tell the deployment of postgres to update if the image is re built?

Reply:
i think that should be a little more manual; to me, the database is sacred. imo its better not to have any automation messing with it without close supervision. i'll build the image, test it out first in a separate deploy, then scale the original deployment down and back up once i am happy with the image.

its cool if we break a deploy with our own code; we broke, we can fix it. but the database is an annoying black box, that if unintentionally broken out of step due to an automated process, can be annoying to fix unexpectedly at an inopportune time.